system-addon/lib/FaviconFeed.jsm

```diff
@@ -134,6 +241,14 @@ this.FaviconFeed = class FaviconFeed {
         Services.scriptSecurityManager.getSystemPrincipal()
       );
     }
+
+    if (!fetchedByTippyTop) {
+      const md5 = md5Hash(url);
+      if (!this._queryForRedirects.has(md5)) {
+        this._queryForRedirects.add(md5);
```

Comment:
Curious, why complicate with hashing instead of just checking/adding the url?

Reply:
Because this will sit in memory, just a preventive move in case the size of URL goes too far (Places caps it as [65536](https://searchfox.org/mozilla-central/source/toolkit/components/places/PlacesUtils.jsm#193)).

Reply:
We have plenty of urls and copies of this exact url in various in-memory caches, etc. And now we have an additional jsm to load in to memory… oh well

Reply:
Oh, I didn't know that we wanted to limit the number of jsm modules. Let me revert this part of change then.